### PR TITLE
Initial streamz docker image to lower barrier for users accessing str…

### PR DESCRIPTION
**Pull request by jdye64**
…eamz exmaples. Closes: #287

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -2,3 +2,5 @@ tornado
 toolz
 zict
 six
+numpy
```
Comment (CJ-Wright) [inline requirements.txt]: These are optional deps I think so we might consider if they are needed or not.

Comment (jdye64) [reply requirements.txt]: That is a good point. Let me remove them from requirements.txt and rather just install them with pip directly in the Dockerfile itself. 

If the conda PR merges at another point that could also be used instead to make the Dockerfile even cleaner.

**Comment by jdye64** [comment]
@CJ-Wright I modified the Dockerfile to use the conda environment from the optional dependencies now instead of pip. This also ensures that as things are added to the conda environment file over time they will be automatically added to the Docker container at build time.

I also added some documentation on building in docs/index.rst. Let me know what you think.

**Comment by martindurant** [comment]
@CJ-Wright , were you reviewing this?

**Comment by martindurant** [comment]
ping @CJ-Wright

**Comment by CJ-Wright** [comment]
Sorry for the delay

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+USER root
+
+ENV DEBIAN_FRONTEND noninteractive
+ENV SCALA_VERSION 2.11
+ENV KAFKA_VERSION 2.3.0
+ENV KAFKA_HOME /opt/kafka_"$SCALA_VERSION"-"$KAFKA_VERSION"
+
+# Install system dependencies and convenience utilities
+RUN apt-get update -y && \
+    apt-get install -y wget \
```
Comment (CJ-Wright) [inline Dockerfile]: Is there any reason not to get these two from conda-forge?

**Comment by CJ-Wright** [comment]
This looks good to me, one minor comment.

**Comment by jdye64** [comment]
Sorry for the major delay folks. Yep agree its dumb of my to not just add them via conda. I'll make that change and commit now after a validation build.

**Comment by CJ-Wright** [comment]
cycle for ci

**Comment by CJ-Wright** [comment]
This is failing on flake8 I'll push a fix to the branch

**Comment by CJ-Wright** [comment]
I don't know why flake8 is unhappy some of the offending lines don't even exist at the most recent commit.

**Comment by martindurant** [comment]
I see no reason not the merge this - @CJ-Wright , agreed?

**Pull request merged**